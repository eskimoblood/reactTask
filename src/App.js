import React, { Component } from 'react';
import { connect } from 'react-redux';
import Content from './Content';
import { startSearch } from './actions'

export class App extends Component {

  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this)
  }

  onChange(event) {
    if (this.timer) {
      window.clearTimeout(this.timer);
    }
    this.timer = window.setTimeout(this.props.startSearch, 250, event.target.value);
  }

  render() {
    return (
      <div>
        <input onInput={this.onChange} />
        <Content {...this.props} />
      </div>
    );
  }
}
const mapStateToProps = (state) => state;

const mapDispatchToProps = dispatch => ({ startSearch: (term) => dispatch(startSearch(term)) });

export default  connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
