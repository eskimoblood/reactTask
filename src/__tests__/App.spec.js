import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { App } from '../App';

jest.mock('../Content', () => 'Content');
jest.useFakeTimers();

describe('App', () => {

  it('renders correct', () => {
    const app = shallow(<App onChange={jest.fn()} />);
    expect(shallowToJson(app)).toMatchSnapshot();
  });

  it('triggers the search', () => {
    const onChange = jest.fn();
    const app = shallow(<App startSearch={onChange} />);
    app.find('input').first().prop('onInput')({ target: { value: 'test123' } });
    jest.runAllTimers();
    expect(onChange).toHaveBeenCalledWith('test123');
  });
});
