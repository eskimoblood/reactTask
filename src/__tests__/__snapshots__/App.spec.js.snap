exports[`App renders correct 1`] = `
<div>
  <input
    onInput={[Function]} />
  <Content
    onChange={[Function]} />
</div>
`;
